server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -1194,6 +1160,32 @@ public void testStatsEnabledIgnoreFailure() throws Exception {
             (pf, t, i, f, c, ctx) -> throwingResponseProcessor
         );
 
+        SearchPipelineService searchPipelineService = getSearchPipelineService(
+            requestProcessorsEnableIgnoreFailure,
+            responseProcessorsEnableIgnoreFailure
+        );
+
+        SearchRequest request = new SearchRequest();
+        SearchResponse response = new SearchResponse(null, null, 0, 0, 0, 0, null, null);
+
+        searchPipelineService.resolvePipeline(request.pipeline("good_request_pipeline")).transformResponse(response);
+        // Caught Exception here
+        searchPipelineService.resolvePipeline(request.pipeline("bad_request_pipeline")).transformResponse(response);
+        searchPipelineService.resolvePipeline(request.pipeline("good_response_pipeline")).transformResponse(response);
+        // Caught Exception here
+        searchPipelineService.resolvePipeline(request.pipeline("bad_response_pipeline")).transformResponse(response);
+
+        // when ignoreFailure enabled, the search pipelines will all succeed.
+        SearchPipelineStats stats = searchPipelineService.stats();
+        assertPipelineStats(stats.getTotalRequestStats(), 2, 0);
+        assertPipelineStats(stats.getTotalResponseStats(), 2, 0);
+
```

Comment:
The only thing left to check in this test is that the "bad" pipelines each have a failure metric count of 1. They didn't throw an exception, but we still count their failures.

Reply:
Make sense. The 2 requests succeeded and 2 responses succeeded but there are one failed request processor and one failed response processor. Will add the additional match for each processor.

Reply:
addressed comment in the new commit [Add test on matching each processor stat](https://github.com/opensearch-project/OpenSearch/pull/8373/commits/73841af86d86a9ed79d9c78d00bf2aba5ad52ee9)